### PR TITLE
feat(web): add share link on asset-viewer

### PR DESCRIPTION
**Pull request by samholton**
Adds a sharing link on the individual asset viewer. This matches the functionality on mobile to share a single image rather than going back to grid view and selecting it. Mentioned in https://github.com/immich-app/immich/discussions/7561 as well.

**before**
before

**after**
after

after with menu

### CODE REVIEW

File: web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte [hunk 1]
```diff
@@ -139,6 +142,14 @@
         title="Download"
       />
     {/if}
+    {#if showShareButton}
```
Comment (martabal) [inline web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte:145]: Can you restrict it to asset owners ?

Comment (samholton) [reply web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte:145]: Initially I included restriction on owner but tested selecting multiple images from the grid view including those that were from partner share. The partner photos showed up in the share link as well. Should a user be able to share partner share photos?

Comment (jrasm91) [reply web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte:145]: I think owner or partner can both share it

**Review by jrasm91 — APPROVED** [review]
This looks fine to me, although I think shared should be the first icon. Also there are a lot of actions now. Maybe we should move download into the overflow menu now.

**Comment by samholton** [comment]
> This looks fine to me, although I think shared should be the first icon. Also there are a lot of actions now. Maybe we should move download into the overflow menu now.

Do we think users would more often download a single image or share a single image?

**Comment by jrasm91** [comment]
It is how Google does it I believe.

**Review by danieldietzler — APPROVED** [review]
LGTM!

File: web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte (hunk 1 above)
Comment (martabal) [reply web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte]: Yep, but that's an issue for assets in shared albums
image

Comment (samholton) [reply web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte]: I'll take a look at assets in shared albums

Comment (samholton) [reply web/src/lib/components/asset-viewer/asset-viewer-nav-bar.svelte]: I was able to reproduce by inviting a user to the shared album that was also _not_ a partner share. The same error is displayed if selecting the image from the main album page and selecting share as well. So seems like an existing but not necessarily related to this change.

Created https://github.com/immich-app/immich/issues/7604 for tracking

**Pull request merged**